test/python/transpiler/test_echo_rzx_weyl_decomposition.py

```diff
@@ -0,0 +1,202 @@
+
+        unitary_after = qi.Operator(after).data
+
+        self.assertTrue(np.allclose(unitary_circuit, unitary_after))
+
+        self.assertRZXgates(unitary_circuit, after)
+
+    def assertRZXgates(self, unitary_circuit, after):
+        alpha = TwoQubitWeylDecomposition(unitary_circuit).a
+        beta = TwoQubitWeylDecomposition(unitary_circuit).b
+        gamma = TwoQubitWeylDecomposition(unitary_circuit).c
+
+        # check whether after circuit has correct number of rzx gates
+        expected_rzx_number = 0
+        if not alpha == 0:
+            expected_rzx_number += 2
+        if not beta == 0:
+            expected_rzx_number += 2
+        if not gamma == 0:
+            expected_rzx_number += 2
+
+        circuit_rzx_number = QuantumCircuit.count_ops(after)["rzx"]
+
+        self.assertEqual(expected_rzx_number, circuit_rzx_number)
+
+    def test_weyl_unitaries_random_circuit(self):
+        """Weyl decomposition for random two-qubit circuit."""
+        theta = pi / 9
+        epsilon = 5
+        delta = -1
+        eta = 0.2
+        qr = QuantumRegister(2, "qr")
+        circuit = QuantumCircuit(qr)
+
+        # random two-qubit circuit
+        circuit.rzx(theta, 0, 1)
+        circuit.rzz(epsilon, 0, 1)
+        circuit.rz(eta, 0)
+        circuit.swap(1, 0)
+        circuit.h(0)
+        circuit.rzz(delta, 1, 0)
+        circuit.swap(0, 1)
+        circuit.cx(1, 0)
+        circuit.swap(0, 1)
+        circuit.h(1)
+        circuit.rxx(theta, 0, 1)
+        circuit.ryy(theta, 1, 0)
+        circuit.ecr(0, 1)
+
+        unitary_circuit = qi.Operator(circuit).data
+
+        dag = circuit_to_dag(circuit)
+        pass_ = EchoRZXWeylDecomposition(self.inst_map)
+        after = dag_to_circuit(pass_.run(dag))
+
+        unitary_after = qi.Operator(after).data
+
+        self.assertTrue(np.allclose(unitary_circuit, unitary_after))
+
+    def test_weyl_parameters(self):
+        """Computation of the correct RZX Weyl parameters"""
+        theta = pi / 3
+        qr = QuantumRegister(2, "qr")
+        circuit = QuantumCircuit(qr)
+        qubit_pair = (qr[0], qr[1])
+
+        circuit.rzz(theta, qubit_pair[0], qubit_pair[1])
+        circuit.swap(qubit_pair[0], qubit_pair[1])
+        unitary_circuit = qi.Operator(circuit).data
+
+        # Weyl parameters (alpha, beta, gamma)
+        alpha = TwoQubitWeylDecomposition(unitary_circuit).a
+        beta = TwoQubitWeylDecomposition(unitary_circuit).b
+        gamma = TwoQubitWeylDecomposition(unitary_circuit).c
+
+        # RZX Weyl parameters (rzx_alpha, rzx_beta, rzx_gamma)
+        rzx_alpha = TwoQubitWeylEchoRZX(
+            unitary_circuit, inst_map=self.inst_map, qubit_pair=qubit_pair
+        ).a
+        rzx_beta = TwoQubitWeylEchoRZX(
+            unitary_circuit, inst_map=self.inst_map, qubit_pair=qubit_pair
+        ).b
+        rzx_gamma = TwoQubitWeylEchoRZX(
+            unitary_circuit, inst_map=self.inst_map, qubit_pair=qubit_pair
+        ).c
```

Comment:
Due to the suggestions above you will also need to adapt the is_native here in the tests.

Reply:
See a484c9c.